tests/chart_tests/test_git_sync_relay_deployment.py

```diff
@@ -181,3 +181,52 @@ def test_gsr_deployment_without_ssh_credentials_and_known_hosts(self, kube_versi
             "--root=/git",
         ]
         assert c_by_name["git-daemon"]["livenessProbe"]
+
+    def test_gsr_deployment_with_resource_overrides(self, kube_version):
+        """Test that gitsync  deployment are configurable with custom resource limits."""
+        resources = {
+            "requests": {"cpu": 0.1, "memory": "512Mi"},
+            "limits": {"cpu": 0.1, "memory": "512Mi"},
+        }
+        values = {
+            "gitSyncRelay": {
+                "enabled": True,
+                "gitSyncResources": resources,
+                "gitDaemonResources": resources,
+            }
+        }
+
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only="templates/git-sync-relay/git-sync-relay-deployment.yaml",
+            values=values,
+        )
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "Deployment"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "release-name-git-sync-relay"
+        c_by_name = {
+            c["name"]: c for c in doc["spec"]["template"]["spec"]["containers"]
+        }
```

Comment:
We should add the get_containers_by_name function to this repo so we're not using this code snip everywhere. https://github.com/astronomer/astronomer/blob/da36df6cc8/tests/__init__.py#L14-L32

Reply:
Yes thought the same

Reply:
adding it

Reply:
took reference from astronomer/astronomer and replaced the changes here